### PR TITLE
Call an optional start-hook script from /bin/start

### PR DESCRIPTION
**Pull request by zen-dog**
An optional start hook script defined via $MARATHON_APP_START_HOOK environment variable is called from /bin/start. This provides a possibility for additional initialization steps when starting marathon.

### CODE REVIEW

**Comment by zen-dog** [comment]
retest this please

File: bin/start [hunk 1]
```diff
@@ -40,6 +40,19 @@ require_arg() {
   fi
 }
 
+# Run the start-hook script [optional]
+if [ -f "$MARATHON_APP_START_HOOK" ]; then
```
Comment (jdef) [inline bin/start]: maybe check with `-x` instead?

**Comment by jdef** [comment]
environment variables that start with `MARATHON_` are also interpreted as marathon flags I think. is it standard practice to use envvars for things that are not also command line flags for Marathon? /cc @aquamatthias

**Comment by aquamatthias** [comment]
`MARATHON_*` is translated to command line flags, but ignores `MARATHON_APP*` so I don't see a conflict with that name. 

`MARATHON_APP_*` is used to set values of the app definition, the task is derived from.

So we probably should name this env var in a way it does not start with `MARATHON_`
E.g.: `HOOK_MARATHON_START`

File: bin/start (hunk 1 above)
Comment (zen-dog) [inline bin/start]: Yeh, makes sense.

**Comment by zen-dog** [comment]
> MARATHON_APP_* is used to set values of the app definition, the task is derived from.
>... E.g.: HOOK_MARATHON_START
I agree, however any name I could come up with that is not starting with MARATHON_ somehow sounds wrong. And since there is no logic that would convert **all** MARATHON_APP* prefixed variables to flags it felt ok to make an exception.  But I can change it if you insist.

**Comment by aquamatthias** [comment]
🚢 it

**Comment by jdef** [comment]
retest this please

**Comment by aquamatthias** [comment]
retest this please

**Pull request merged**